prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -15,19 +17,46 @@
 
 logger = getLogger(__name__)
 
+# pylint: disable=too-many-instance-attributes
+
 
 class RedfishHelper:
     """Helper function for redfish."""
 
     def __init__(self, config: Config) -> None:
-        """Initialize redfish login args and cache TTL value for discover method."""
+        """Initialize values for class"""
         self.host = config.redfish_host
         self.username = config.redfish_username
         self.password = config.redfish_password
         self.timeout = config.redfish_client_timeout
         self.max_retry = config.redfish_client_max_retry
         self.discover = self.get_discover(config.redfish_discover_cache_ttl)
 
+        self.sys_inventory: Optional[List[Dict[str, Any]]] = None
+        self.redfish_obj: Optional[HttpClient] = None
+
+    def _get_redfish_obj(self) -> Optional[HttpClient]:
```

Comment:
**issue:** 

Return None is not a nice practice of error handling.

Reply:
The purpose are very different here.
In the original function, redfish_obj maybe none is because we use it to control if we need to logout in the end, in the same function.

But `_get_redfish_obj` doesn't include this logic, so user may want to know the actual error instead of just None.

Reply:
This has been fixed by having the exception be raised directly from redfish_client in the `__enter__` method of RedfishHelper. This is caught while using the context manager in RedfishCollector.